### ISSUE TITLE
Create the "Most popular content by pageviews" key metric widget tile

### ISSUE THREAD

**Issue by jimmymadon** [google/site-kit-wp#6247]
## Feature Description

<img width="384" alt="Screenshot 2022-12-19 at 01 37 41" src="https://user-images.githubusercontent.com/1721319/210283482-e2e9cbd8-ac58-4857-8147-6a90b647dbfd.png">

- Refer to the widget tables in the [design doc](https://docs.google.com/document/d/1k8NOo2xEHUQAcgYG0TB5sUjGcPiC8SWIalEdd9G8-9g/edit#bookmark=id.qz6oxctz8fe7) and the [PRD](https://docs.google.com/document/d/1JvJ2gKxaX2PcYOnmA3-xqz4hK_ZHxj8Rd0svq6S8tiU/edit#bookmark=id.w8o73hf5pz1f) explaining the sources of data and other details.
- [Figma mock for all widgets](https://www.figma.com/file/CdtrHB0HAjfGRWTnTMMQI0/Key-Metrics?node-id=3407%3A73377&t=mxlhDsRNGXcli8tk-0)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A new component should be created for the 'Most popular content by pageviews' Key metric widget and should be displayed within the Key Metrics widget area if it is selected (based on #6257 and #6313).
* The titles of the top three most viewed pages should be displayed from the [Google Analytics Data API](https://developers.google.com/analytics/devguides/reporting/data/v1/api-schema) based on the following metric/dimension and the global date range selected on the Site Kit dashboard:
    * metric: `screenPageViews`
    * dimension: `pagePathPlusQueryString`
* The total `screenPageViews` for the top 3 pages should be displayed next to their respective titles as per the Figma mock.
* Existing GA4 datapoints and datastore infrastructure that is being added as part of the [GA4 Reporting Epic](https://github.com/google/site-kit-wp/issues/6253#workspaces/execution-5e185a810bd7fc358af69885/board?epics=Z2lkOi8vcmFwdG9yL0VwaWMvMTAxNTQyOQ) should be used where available.

## Implementation Brief

* Once #6244 has been merged.
* In `assets/js/modules/analytics-4/components/widgets/PopularContentWidget.js`:
  * Return `null` if the `GA4` module is not connected.
  * Using the `getKeyMetrics` selector, check if any of the selected Metric is from GA4. If not, return `null`. Otherwise, continue.
  * Most of the styles should already be in place, with #6244 and #6251. Reuse the `MetricTileTable` component.
  * To get the data for the component, use the following options:
      * metrics: `screenPageViews`
      * dimensions: `pageTitle` and `pagePath`
      * limit: `3`
      * orderBy: `screenPageViews` (descending)
* Add the styles for the tabular data section in the `assets/sass/components/key-metrics` folder.
    * Instead of using table cells, use flexbox to achieve the tabular layout, with `flex-wrap` set to `none`.
    * The string truncation should be done via CSS only (`text-overflow: ellipsis;`)
* Refer to `assets/js/components/DetailsPermaLinks.js` about how to get the full URL for page paths.
* Add stories for `AnalyticsPopularContentWidget` in `assets/js/modules/analytics-4/components/widgets/AnalyticsPopularContentWidget.stories.js`.
* Look into the possibility to having a `assets/js/modules/analytics-4/components/common/RowWithNumericMetric.js` component which accepts a `label` (string or component) and `value` as props which can be re used by other components. The `RowWithNumericMetric ` component will be responsible to generate the a single row of data within the table along with the required styles.
    * Screenshot for reference: <img width="513" alt="image" src="https://user-images.githubusercontent.com/31818/217196304-ebccf5ca-f815-42f6-b6b9-82f285204ebe.png">

### Test Coverage
* <del>Add tests for the newly added components.</del>

## QA Brief

* Same as for the #6244 ticket:
  * With the `userInput` feature flag enabled, answer the questionnaire. For now, all widgets are displayed on the dashboard (which will change with future commits). As part of this issue, test the design and metric values of the `Most popular content by pageviews` widget rendered in the Key Metrics section.
  * Test the generic `Most popular content by pageviews` stories within the Key Metrics widget section in Storybook.
  * Note: Designing the error state for widget tiles is out of the scope of this issue. A new issue is or will be created for this.

## Changelog entry

* Create the "Most popular content by pageviews" key metric widget tile.

**Comment by tofumatt** [comment]
ACs 👍🏻

**Comment by tofumatt** [comment]
> Look into the possibility to having a `assets/js/modules/analytics-4/components/common/Info.js` component which accepts a `label` (string or component) and `value` as props which can be re used by other components. The `Info` component will be responsible to generate the a single row of data within the table along with the required styles.

I'm suspicious of "info" and "data" as descriptors/variable names/component names, because the naming doesn't really describe what they're doing or showing. It's so generic as to usually be confusing 🤔 

In this case that's happening for me—what section of this new component would this component render? It might be helpful to highlight/outline it visually in a screenshot, but I'm a bit lost as to what this is.

The rest of the IB looks good, but this bit has me stumped 😅

**Comment by asvinb** [comment]
@tofumatt I've updated the IB to include the screenshot. Actually I was trying to find a better name for `Info` but I couldn't. Any suggestions?

**Comment by tofumatt** [comment]
@asvinb Thanks for the screenshot!

In this case I'd say an overly-verbose/specific name beats out the very generic. I'd go with something like `RowWithNumericMetric` or something like that. But even just `RowWithNumericMetric` or `TitleAndNumericMetric` would work.

Is that too specific to this one case or do you think it's generic enough of a name to work?

**Comment by asvinb** [comment]
That sounds good @tofumatt . IB updated!

**Comment by hussain-t** [comment]
@asvinb, the KM widget components (skeleton) are created in #6313. So we don't have to create any components.

**Comment by hussain-t** [comment]
@asvinb @tofumatt, according to the AC, the criteria to render the widget tile is missing in the IB:
>A new component should be created for the 'Most popular content by pageviews' Key metric widget and should be displayed within the Key Metrics widget area if it is selected (based on https://github.com/google/site-kit-wp/issues/6257

It should check if the selected metric depends on GA4 using the `getKeyMetrics` selector and render the elements.

**Comment by tofumatt** [comment]
@hussain-t Thanks for pointing that out, that makes sense! Can you add it to the IB and then I can review it/more it to the backlog? I think that's all the change that's needed here, right?

**Comment by hussain-t** [comment]
Thanks, @tofumatt. I updated the IB with a couple of changes.

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

@eugene-manuilov 

**Issue 1:** In case of zero data. The message is getting overflow from tile, and it is happening with other tile "Top Performing Keywords" also. This issue was not existed before, as you can screenshot in my comment under #6251. Should I create new ticket for this, or you will fix this in follow up PR ?

![image](https://github.com/google/site-kit-wp/assets/94359491/5796a77d-7e05-4418-9906-3473671e76e7)
 

**Question 2** : I've noticed difference between "Most Popular content by pageviews" tile and "Top content over the last 90 days" widget page views data. It is showing different when I'm selecting date range for 90 days. It is showing similar when I'm selecting 7 days or 14 days and with 1 number difference when I'm selecting date range 28 days. I'm expecting that tile and widget page views  will match in all date selection.

<details>

90 days

![image](https://github.com/google/site-kit-wp/assets/94359491/98e85575-0932-4d04-9293-df2694f72b66)

![image](https://github.com/google/site-kit-wp/assets/94359491/abe5bd77-411f-431f-8afe-9a4d3fb0bdea)

28 days 

![image](https://github.com/google/site-kit-wp/assets/94359491/8b84ffe4-603f-4041-b1ae-99a3b9957cad)

![image](https://github.com/google/site-kit-wp/assets/94359491/b6a6d017-5d5b-4de5-a1b7-960455e78299)

</details>

**Comment by mohitwp** [comment]
###vQA Update ❌ 

As discussed in Sprint call @eugene-manuilov will create follow-up PR for issues reported [above](https://github.com/google/site-kit-wp/issues/6247#issuecomment-1611067084)

**Comment by aaemnnosttv** [comment]
This is ready for another pass @mohitwp 👍

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on main environment.
- Verified the "Most popular content by pageviews" key metric widget tile with Figma design.
- Issue Number 1 reported [above](https://github.com/google/site-kit-wp/issues/6247#issuecomment-1611067084) is now resolved.
- For the Issue 2 data difference, we have separate ticket - https://github.com/google/site-kit-wp/issues/7214.
- Verified the storybook story `Most popular content by pageviews` stories.

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/fc5d1a81-ea26-4221-aa1c-943e5b3d8778)

![image](https://github.com/google/site-kit-wp/assets/94359491/49d52977-4242-4571-9358-3ff42b8a6649)

</details>
